src/test/java/io/split/android/client/storage/attributes/AttributesStorageImplTest.java

```diff
@@ -0,0 +1,200 @@
+        attributesStorage = new AttributesStorageImpl(null);
+
+        attributesStorage.loadLocal();
+    }
+
+    @Test
+    public void clearCallsClearInPersistentStorageIfPersistentStorageIsNotNull() {
+        attributesStorage.clear();
+
+        Mockito.verify(persistentAttributesStorage).clear();
+    }
+
+    @Test
+    public void clearDoesNotCallClearInPersistentStorageWhenItIsNull() {
+        attributesStorage = new AttributesStorageImpl(null);
+
+        attributesStorage.clear();
+    }
+
+    @Test
+    public void getFetchesCorrespondingValuePreviouslyLoadedFromPersistentStorageIfPersistentStorageIsNotNull() {
+        Map<String, Object> defaultValuesMap = getDefaultValuesMap();
+
+        Mockito.when(persistentAttributesStorage.getAll()).thenReturn(defaultValuesMap);
+
+        attributesStorage.loadLocal();
+
+        assertEquals("value1", attributesStorage.get("key1"));
+    }
+
+    @Test
+    public void getReturnsNullIfValueIsNotPresent() {
+        Map<String, Object> defaultValuesMap = new HashMap<>();
+
+        Mockito.when(persistentAttributesStorage.getAll()).thenReturn(defaultValuesMap);
+
+        attributesStorage.loadLocal();
+
+        assertNull(attributesStorage.get("key1"));
+    }
+
+    @Test
+    public void getAllReturnsAllValuesPreviouslyRetrievedFromPersistentStorageIfPersistentStorageIsNotNull() {
+        Map<String, Object> defaultValuesMap = getDefaultValuesMap();
+
+        Mockito.when(persistentAttributesStorage.getAll()).thenReturn(defaultValuesMap);
+
+        attributesStorage.loadLocal();
+
+        final Map<String, Object> allAttributes = attributesStorage.getAll();
+
+        assertEquals(defaultValuesMap, allAttributes);
+    }
+
+    @Test
+    public void setUpdatesValueInMemory() {
+        attributesStorage.set("newKey", 200);
+
+        assertEquals(200, attributesStorage.get("newKey"));
+    }
+
+    @Test
+    public void setUpdatesValueInPersistentStorageIfPersistentStorageIsNotNull() {
+        Map<String, Object> expectedMap = new HashMap<>();
+        expectedMap.put("newKey", 200);
+        ArgumentCaptor<Map<String, Object>> captor = ArgumentCaptor.forClass(Map.class);
+
+        attributesStorage.set("newKey", 200);
+
+        Mockito.verify(persistentAttributesStorage).set(captor.capture());
+        assertEquals(expectedMap, captor.getValue());
+    }
+
+    @Test
+    public void setDoesNotInteractWithPersistentStorageWhenItIsNull() {
+        attributesStorage = new AttributesStorageImpl(null);
+
+        attributesStorage.set("newKey", "newValue");
+    }
+
+    @Test
+    public void setWithMapUpdatesMultipleValuesInMemory() {
+        Map<String, Object> defaultValuesMap = getDefaultValuesMap();
+        attributesStorage.set(defaultValuesMap);
+
+        assertEquals(defaultValuesMap, attributesStorage.getAll());
+    }
+
+    @Test
+    public void setWithMapUpdatesMultipleValuesInPersistentStorageIfPersistentStorageIsNotNull() {
```

Comment:
Same that in line 79. I think it would be more easy to understand what the test does.